adam_core/utils/tests/test_indexable.py

```diff
@@ -0,0 +1,470 @@
+from copy import deepcopy
+
+import numpy as np
+import pytest
+from astropy.time import Time
+
+from ..indexable import (
+    Indexable,
+    _convert_grouped_array_to_slices,
+    _map_values_to_integers,
+    concatenate,
+)
+
+SLICES = [
+    slice(0, 1, 1),
+    slice(1, 2, 1),
+    slice(8, 9, 1),
+    slice(7, 8, 1),
+    slice(0, 10, 1),
+    slice(0, 5, 1),
+    slice(5, 10, 1),
+    slice(50, 60, 2),
+    slice(50, 60, 2),
+    slice(99, 10, -2),
+]
+
+ATTRIBUTES = [
+    "index_array_int",
+    "index_array_str",
+    "array",
+    "array_2d",
+    "array_3d",
+    "masked_array",
+    "times",
+]
+N = 100
+
+
+class TestIndexable(Indexable):
+    def __init__(self):
+
+        self.index_array_int = np.concatenate(
+            [np.arange(10, dtype=int) for i in range(10)]
+        )
+        self.index_array_str = np.concatenate(
+            [np.arange(10, dtype=int).astype(str) for i in range(10)]
+        )
+        self.array = np.arange(0, N)
+        self.array_2d = np.random.random((N, 6))
+        self.array_3d = np.random.random((N, 6, 6))
+        self.masked_array = np.ma.arange(0, N)
+        self.masked_array.mask = np.zeros(N)
+        self.masked_array.mask[0:N:2] = 1
+        self.times = Time(np.arange(59000, 59000 + N), scale="utc", format="mjd")
+
+        Indexable.__init__(self)
+
+
+def assert_equal(a, b):
+    if isinstance(a, (np.ndarray)) and isinstance(b, (np.ndarray)):
+        np.testing.assert_equal(a, b)
+    elif isinstance(a, (np.ma.masked_array)) and isinstance(b, (np.ma.masked_array)):
+        np.testing.assert_equal(a.data, b.data)
+        np.testing.assert_equal(a.mask, b.mask)
+    elif isinstance(a, (Time)) and isinstance(b, (Time)):
+        np.testing.assert_equal(a.mjd, b.mjd)
+        assert a.scale == b.scale
+        assert a.format == b.format
+    else:
+        assert a == b
+
+    return
+
+
+def test__convert_grouped_array_to_slices_raises():
+    # Test that the function raises a ValueError if the array is not grouped
+    # (sorted but sort order does not matter)
+    array = np.concatenate([np.arange(0, 5), np.arange(0, 5)])
```

Comment:
nit: I'd find it clearer to spell out the array instead of concatenating function calls. For example:
```suggestion
    array = np.array([0, 1, 0, 1])
```

Maybe that's just because I don't use numpy as much, but I find it more legible.

Reply:
Done!